src/app/(admin)/new_dashboard/guides/hooks/guideClientHooks.ts

```diff
@@ -0,0 +1,63 @@
+import { GuideClient } from "@/api/guide.client";
+import { useQuery, useMutation } from "@tanstack/react-query";
+import { CreateReviewDto } from "../dto/CreateReviewDto";
+import {
+  Guide,
+  GuideRequestUpdate,
+  UpdateGuidePayload,
+} from "@/api/types/Guide";
+import { useRouter } from "next/navigation";
+
+const guideClient = new GuideClient();
+
+export function useAllGuides() {
+  return useQuery({
+    queryKey: ["guides"],
+    queryFn: () => guideClient.getAllGuides(),
+  });
+}
+
+export function useGuide(guideId: number) {
+  return useQuery({
+    queryKey: ["guide", guideId],
+    queryFn: () => guideClient.getGuide(guideId),
+  });
+}
+
+export function useDeleteGuide(guideId: number) {
+  const router = useRouter();
+  return useMutation({
+    mutationFn: () => guideClient.deleteGuide(guideId),
+    onSuccess: () => {
+      router.push("/new_dashboard/guides");
+    },
+    onError: () => {
+      alert("Failed to delete guide. Please try again.");
+    },
```

Comment:
Why is there an alert here? We should display an alert with mui. There are lot of components that can be used to display this